src/pages/docs/tutorials/tutorial-typescript-enact/adding-a-new-component/index.md

```diff
@@ -0,0 +1,97 @@
+---
+title: Add a New Component Using TypeScript
+github: https://github.com/enactjs/docs/blob/develop/src/pages/docs/tutorials/tutorial-typescript-basic/app-setup/index.md
+order: 3
+---
+
+### Creating a Counter Component
+
+- Create a **Counter** folder in **src/components** and add a **Counter.tsx** file
+
+```none
+./src/components/Counter/Counter.tsx
+```
+
+### Counter using Typescript
+
+We can now populate the **Counter.tsx** file with a simple counter and add the following contents:
+
+```ts
+import * as React from 'react';
+import Button from '@enact/moonstone/Button';
+
+export default class Counter extends React.Component {
+	state = {
+	  count: 0
+	};
+
+	increment = () => {
+	  this.setState({
+		count: (this.state.count + 1)
+	  });
+	};
+
+	decrement = () => {
+	  this.setState({
+		count: (this.state.count - 1)
+	  });
+	};
+	reset = () => {
+		this.setState ({
+			count: 0
+		})
+	}
+
+	render () {
+	  return (
+		<div>
+		  <h1>{this.state.count}</h1>
+		  <Button onClick={this.decrement}>Decrement --</Button>
+		  <Button onClick={this.reset}>Reset</Button>
+		  <Button onClick={this.increment}>Increment ++</Button>
+		</div>
+	  );
+	}
+  }
+
+export default Counter;
+```
+
+You'll also need a **package.json** in the same directory to indicate the module's entry point:
+
+```json
+{
+    "main": "Counter.tsx"
+}
+```
+
+Then, inside of **views/MainPanel.tsx**, we can load the `Counter` component:
+
+```ts
+//Custom component
+import Counter from '../components/Counter';
+
+const MainPanel = kind({
+    name: 'MainPanel',
+
+    render: (props) => (
+        <Panel {...props}>
+            <Header title="Hello Enact + TypeScript!" />
+            <Counter />
+        </Panel>
+    )
+});
+
+export default MainPanel;
+```
+
+- Run the App in the terminal
+
+```bash
+npm run serve
+```
+### Typescript Counter in the Browser
+
+![Typescript Simple Counter](counter_view.png)
```

Comment:
The screenshot is trying to load from localhost and it doesn’t exist.  Not sure if this is a doc issue, user issue, or will go away when it’s deployed to the docs site, but we should check into it.